flake.nix

```diff
@@ -0,0 +1,28 @@
+{
+  description = "A flake defining upgrade-provider build-from-source package";
+
+  inputs = {
+    nixpkgs.url = github:NixOS/nixpkgs/nixos-23.05;
+  };
+
+  outputs = { self, nixpkgs }: let
+
+    package = { system }: let
+      pkgs = import nixpkgs { system = system; };
+    in pkgs.buildGo120Module rec {
+      name = "upgrade-provider";
+      version = ''${self.rev or "dirty"}'';
+      src = ./.;
+      # subPackages = [ "cmd/pulumictl" ];
+      doCheck = false;
+      vendorSha256 = "sha256-YBteVgcWvIE10ojd9W4grMK8kJ0zXXQiBuEHto3sABI=";
```

Comment:
Will we have to update this any time the source changes?

Reply:
Yeah, unfortunately this needs to get updated any time go.sum changes. I've not found yet a nix-native way to make his part of the flake.lock to automatically update.

Reply:
It's not affected by source changes though. When this needs editing, you run `nix build` and it suggests the new sha256 to insert here.